tools/o3de-doxygen/README.md

```diff
@@ -0,0 +1,184 @@
+
+### Frameworks API
+
+In the `o3de-doxygen` directory, run the script:
+
+```shell
+./o3de-framework-apis.sh
+```
+
+This generates the following files in `o3de.org`: 
+ - `content/docs/api/frameworks/_index.md`
+ - `static/docs/api/frameworks/*`
+
+### Gems API
+
+*Note: You must [install rust commands](#install-rust-commands) to run `o3de-gem-apis.sh`.*
+
+In the `o3de-doxygen` directory, run the script:
+
+```shell
+./o3de-gem-apis.sh
+```
+
+This generates the following files in `o3de.org`: 
+  - `content/docs/api/gems/_index.md`
+  - `static/docs/api/gems/*`
+
+
+
+## Publish
+
+After generating the C++ API reference and supplemental files, it's time to publish them on the website. When you [cleaned your `o3de.org` clone](#o3deorg), you set up a new clean branch, such as `api-ref`, and when you [generated](#generate) the APIs, the files were placed in that branch. 
+
+To publish the newly generated APIs, you must run `git add`, `git commit`, `git push`, and create a new pull request. For guidance through each step, please refer to the [O3DE Docs Contributions: Runbook](https://www.o3de.org/docs/contributing/to-docs/git-workflow/).
+
+
+## Generating for Gems outside of `o3de`
+
+In some cases, the O3DE API Reference must include the API for code that lives outside of the `o3de` repository. This is typically decided by O3DE members and the Release Special Interest Group (SIG). For example, a major O3DE release may include new Gems that live in the `o3de-extras` repository.
+
+To generate API References for Gems outside of `o3de`: 
+
+1. Move or copy the Gem's directory into the `Gems` directory of your local `o3de` clone. 
+
+    To do this using the terminal in your Linux machine, run the following `cp` command with your specified paths. This copies the Gem into the `o3de/Gems/` directory.
+
+    ```shell
+    cp -r <path-to-gem> <path-to>/o3de/Gems
+    ```
+
+2. [Generate the Gems API](#gems-api). 
+
+
+## Customizing the API's main page
+
+The main page is the first page you see when you open a framework or Gem's Doxygen-generated API reference. For example, open any [framework](https://www.o3de.org/docs/api/frameworks/) or [Gem](https://www.o3de.org/docs/api/gems/) in the API docs. The contents of the main page is either automatically generated, or specified by the developer by using the Doxygen command, [/mainpage](https://doxygen.nl/manual/commands.html#cmdmainpage), in the code's comments.
+
+By default, if `/mainpage` is not specified, an API's main page includes a brief introduction and a table of contents, like the following: 
+
+![Default mainpage example](files/default-mainpage-example.png)
+
+If a developer documents the API using `/mainpage`, it will replace the default introduction. However, you may still insert the table of contents by adding `o3de-doxygen-insert-table` to your `/mainpage` comment block. For example, this comment block in the AQtComponents's source code generates the following main page. 
```

Comment:
```suggestion
If a developer documents the API using `/mainpage`, it will replace the default introduction. However, you may still insert the table of contents by adding `o3de-doxygen-insert-table` to your `/mainpage` comment block. For example, this comment block in the AzQtComponents's source code generates the following main page. 
```